OPHD/States/MapViewStateIO.cpp

```diff
@@ -408,7 +408,7 @@ void MapViewState::readStructures(Xml::XmlElement* element)
 		if (structureId == StructureTypeID::SID_AGRIDOME ||
 			structureId == StructureTypeID::SID_COMMAND_CENTER)
 		{
-			auto& foodProduction = *static_cast<FoodProduction*>(&structure);
```

Comment:
I like seeing this in code and seeing the improvement -- this is so much better than all the casting.

Question though -- since it appears to return a reference, what if this is called on a structure that doesn't have a `FoodProduction` component? Does it throw, barf or something else?

Reply:
In this case we are confident there will be a `FoodProduction` component, and deem it a programming error if that's not true. `getComponent` will throw if the component does not exist. If you have similar logic where you aren't sure whether the structure will have the component or not, you can use the `tryGetComponent` function instead, which returns a pointer that can be `nullptr` if the structure didn't have the queried component.

Reply:
Nice, I like that.